frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -287,6 +301,9 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
       case SidebarContextType.ADD_ASSOCIATED_LAND:
         if (propertyType !== 'land') {
           setPropertyType('land');
+        } else if (!buildingToAssociateLand?.id) {
```

Comment:
if the user somehow loads the associated land form without an attached building, kick them out to the selector since there is no way to "fix" the underlying building object within the associated land form.